reference/outcontrol/user-level-output-buffers.xml

```
@@ -0,0 +1,703 @@
+   or its alias <constant>PHP_OUTPUT_HANDLER_CONT</constant>
+   is set when the size of its contents equals or exceeds the size of the buffer
+   and the handler is invoked while the buffer is being automatically flushed.
+   <constant>PHP_OUTPUT_HANDLER_FLUSH</constant> is set
+   when the handler is invoked by calling <function>ob_clean</function>,
+   <function>ob_end_clean</function> or <function>ob_get_clean</function>.
+   When <function>ob_end_clean</function> or <function>ob_get_clean</function>
+   is called, <constant>PHP_OUTPUT_HANDLER_FINAL</constant> is set as well.
+  </para>
+  <note>
+   <para>
+    When <function>ob_end_flush</function> or <function>ob_get_flush</function>
+    is called, <constant>PHP_OUTPUT_HANDLER_FINAL</constant> is set
+    but <constant>PHP_OUTPUT_HANDLER_FLUSH</constant> is not.
+   </para>
+  </note>
+ </section>
+
+ <section>
+  <title>Output Handler Return Values</title>
+  <para>
+   While the return value of the output handler will be ultimately cast to string
+   the implementation of the handler can return a non-string value.
+   This casting to string of this value mostly follows PHP's type casting rules
+   with a few exceptions.
+  </para>
+  <para>
+   The following types and values follow the regular type casting rules:
+   <itemizedlist>
+    <listitem>
+     <para>
+      &null; - returns an empty string
+     </para>
+    </listitem>
+    <listitem>
+     <para>
+      <type>int</type> and <type>float</type> - return a string
+      (e.g. <literal>"1"</literal>, <literal>"1.1"</literal>)
+     </para>
+    </listitem>
+    <listitem>
+     <para>
+      <type>object</type> - returns a string or throws an exception
+      if <methodname>__toString</methodname> is not implemented
+     </para>
+    </listitem>
+    <listitem>
+     <para>
+      <type>array</type> - returns <literal>"Array"</literal> but
+      does not trigger the <literal>Array to string conversion</literal> warning
```

This seems to be a bug in php-src, and I can guess what C function it calls...

---

Yeah I was imagining that ``convert_to_string()`` was being called. It probably should use the ``try_`` variant instead of silently failing for objects. And ideally check for the return value to be of the correct type and throw a warning/TypeError, but this is php-src improvement territory.